Add release notes for Python agent v13.0.1 (#24179)
newrelic/docs-website: src/content/docs/release-notes/agent-release-notes/python-release-notes/python-agent-130001.mdx

---
subject: 'Python agent'
releaseDate: '2026-05-21'
version: 13.0.1
downloadLink: 'https://pypi.python.org/pypi/newrelic'
features: []

> bugs: ['Fix context progagation in LangChain and LangGraph', 'Fix an OpenAI chat completion recording issue']

Reviewer: ```suggestion
bugs: ['Fix context progagation in LangChain and LangGraph', 'Fix an OpenAI chat completion recording issue', 'Fix an issue when using OpenAI streaming through LiteLLM as a proxy']
```

security: []
---

## Notes

> This release of the Python agent fixes a context propagation issue in [LangChain](https://pypi.org/project/langchain) and [LangGraph](https://pypi.org/project/langgraph) and an [OpenAI](https://pypi.org/project/openai) chat completion recording issue.

Reviewer: ```suggestion
This release of the Python agent fixes a context propagation issue in [LangChain](https://pypi.org/project/langchain) and [LangGraph](https://pypi.org/project/langgraph), an [OpenAI](https://pypi.org/project/openai) chat completion recording issue, and an issue when using OpenAI(https://pypi.org/project/openai) streaming through LiteLLM as a proxy.
```

Install the agent using `easy_install/pip/distribute` via the [Python Package Index](https://pypi.python.org/pypi/newrelic), using `conda` via the [Conda-Forge Package Index](https://anaconda.org/conda-forge/newrelic), or download it directly from the [New Relic download site](https://download.newrelic.com/python_agent/release).

> 

Reviewer 2: ```suggestion

```

## Bugs

> * Fix context progagation in LangChain and LangGraph

Reviewer: ```suggestion
* Fix context propagation in LangChain and LangGraph
```

>   * Adds additional transaction context propagation to [LangChain](https://pypi.org/project/langchain) and [LangGraph](https://pypi.org/project/langgraph) to support a wider range of application structures and reformats [LangChain](https://pypi.org/project/langchain) instrumentation to only allow keyword arguments to complex function signatures to prevent silent mistakes.

Reviewer 2: ```suggestion
  * Adds additional transaction context propagation to [LangChain](https://pypi.org/project/langchain) and [LangGraph](https://pypi.org/project/langgraph) to support a wider range of application structures.
```

This is just internal implementation and doesn't need to be in the release notes.

> * Fix an OpenAI chat completion recording issue
> 
>   * [OpenAI](https://pypi.org/project/openai) reintroduced logic that closes the streaming response even if the consumer hasn't read all data. Previously, the chat completion data was only recorded when reaching the stream's end which resulted in chat completion data not being recorded at the closing of the stream. Logic has now been added to capture the chat completion data even if the stream has not been fully read.
> 
> 

Reviewer 2: ```suggestion

* Fix an issue when using OpenAI streaming through LiteLLM as a proxy

  * Previously when using the [OpenAI](https://pypi.org/project/openai) SDK connected through a [LiteLLM](https://www.litellm.ai/) proxy server it was possible to lose reporting of LLM response data due to a missing finish reason. This has been fixed, and the response should now always be recorded.

```
